### PR TITLE
End of the backend step in WeEat project.

### PR DESCRIPTION
**Pull request by Avisha1**
- Add the association between review and restaurant in reviews_controller and all the corresponding places (like erb files).

- Add limitation in review's view in rating (make it in range of 0 to 3).

### CODE REVIEW

File: app/controllers/reviews_controller.rb [hunk 1]
```diff
@@ -1,63 +1,110 @@
 class ReviewsController < ApplicationController
   before_action :set_review, only: [:show, :edit, :update, :destroy]
 
-  # GET /reviews
-  # GET /reviews.json
+  # GET /restaurants/:restaurant_id/reviews
+  # GET /restaurants/:restaurant_id/reviews.xml
```
Comment (omergertel) [inline app/controllers/reviews_controller.rb]: XML? So hard core 😃 
Why not json?

File: app/controllers/reviews_controller.rb [hunk 2]
```diff
@@ -1,63 +1,110 @@
...
   def index
-    @reviews = Review.all
+    #1st you retrieve the restaurant thanks to params[:restaurant_id]
+    restaurant = Restaurant.find(params[:restaurant_id])
```
Comment (omergertel) [inline app/controllers/reviews_controller.rb]: If you want, you can add `before_action :find_restaurant` at the top of the controller and then add this method somewhere:

```
def find_restaurant
  @restaurant = Restaurant.find(params[:restaurant_id])
end
```

And then `@restaurant` will be available in all the methods.

File: app/views/reviews/_form.html.erb [hunk 3]
```diff
@@ -26,10 +26,10 @@
     <%= form.text_field :date, id: :review_date %>
   </div>
 
-  <div class="field">
-    <%= form.label :restaurant_id %>
-    <%= collection_select(:restaurant, :restaurant_id, Restaurant.all, :id, :name, {:selected => params[:id]}) %>
-  </div>
+  <!--<div class="field">-->
```
Comment (omergertel) [inline app/views/reviews/_form.html.erb]: Usually, it's better to just remove dead code

**Review by omergertel — APPROVED** [review]

File: app/controllers/reviews_controller.rb (hunk 1 above)
Comment (Avisha1) [reply app/controllers/reviews_controller.rb]: very TRUE!

Misused the copy=paste mechanism.

File: app/controllers/reviews_controller.rb (hunk 2 above)
Comment (Avisha1) [reply app/controllers/reviews_controller.rb]: Cool tip thanks :)

File: app/views/reviews/_form.html.erb (hunk 3 above)
Comment (Avisha1) [reply app/views/reviews/_form.html.erb]: TRUE!

**Pull request merged**